diff --git a/arch/arm/dts/rock960-model-c-linux.dts b/arch/arm/dts/rock960-model-c-linux.dts
index 1a9e87c342..a33c995eec 100644
--- a/arch/arm/dts/rock960-model-c-linux.dts
+++ b/arch/arm/dts/rock960-model-c-linux.dts
@@ -274,6 +274,18 @@
 				regulator-boot-on;
 				regulator-name = "vcc33_lcd";
 			};
+
+			vcc1v8_dvp: LDO_REG1 {
+				regulator-name = "vcc1v8_dvp";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+				regulator-boot-on;
+				regulator-state-mem {
+					regulator-on-in-suspend;
+					regulator-suspend-microvolt = <1800000>;
+				};
+			};
 		};
 	};
 };
diff --git a/board/rockchip/evb_rk3399/evb-rk3399.c b/board/rockchip/evb_rk3399/evb-rk3399.c
index b6f730852a..7027d69253 100644
--- a/board/rockchip/evb_rk3399/evb-rk3399.c
+++ b/board/rockchip/evb_rk3399/evb-rk3399.c
@@ -12,6 +12,8 @@
 #include <dm/uclass-internal.h>
 #include <asm/setup.h>
 #include <asm/arch/periph.h>
+#include <asm/gpio.h>
+#include <asm/io.h>
 #include <power/regulator.h>
 #include <u-boot/sha256.h>
 #include <usb.h>
@@ -165,6 +167,14 @@ static void setup_serial(void)
 
 int misc_init_r(void)
 {
+
+	puts("Pcie:  Enable Power\n");
+
+	__raw_writel(0xffff0001, (void __iomem *)0xff77e640);
+
+	gpio_request(69, "pcie-power"); /* GPIO2_A5 */
+	gpio_direction_output(69, 1);
+
 	setup_serial();
 	setup_macaddr();
 
